core/beacon/src/authority.rs

```diff
@@ -124,17 +192,31 @@ impl Authority {
                 }
             }
         }
-        // assert_eq!(dup_proposals.len(), num_seats);
+        assert!(
+            dup_proposals.len() >= num_seats as usize,
+            "Number of selected seats {} < total number of seats {}",
+            dup_proposals.len(),
+            num_seats
+        );
+
+        // Shuffle proposals.
+        let mut seed_usize = vec![];
```

Comment:
@frol is there a better way to do this?

Reply:
Does `seed.as_ref().iter().map(|i| i as usize)` work?